src/Npgsql/NpgsqlCommand.cs

```diff
@@ -581,8 +581,25 @@ Task Prepare(bool async, CancellationToken cancellationToken = default)
             for (var i = 0; i < Parameters.Count; i++)
                 Parameters[i].Bind(connector.TypeMapper);
 
-            ProcessRawQuery(connector.UseConformingStrings, deriveParameters: false);
-            Log.Debug($"Preparing: {CommandText}", connector.Id);
+            _connectorPreparedOn = connector;
+
+            if (connector.PreparedStatementManager.BySql.TryGetValue(CommandText, out var preparedStatement) &&
+                preparedStatement.IsExplicit &&
```

Comment:
The idea here is to convert existing auto-prepared into explicitly-prepared - various state needs to be updated, so we go through the path below rather than this hot path. But the very next time this SQL gets executed, it will be explicitly-prepared so all will be well. The case of transitioning auto->explicit seems rare enough that this doesn't seem to be worth optimizing - makes sense?

Reply:
Right